cse-lib/data/src/main/java/com/farao_community/farao/cse/data/cnec/CracResultsHelper.java

```diff
@@ -291,7 +292,8 @@ public String getOrderCode(NetworkElement networkElement) {
     public FlowCnec getWorstCnec() {
         double worstMargin = Double.MAX_VALUE;
         Optional<FlowCnec> worstCnec = Optional.empty();
-        for (FlowCnec flowCnec : crac.getFlowCnecs()) {
+        Set<FlowCnec> optimizedFlowCnecs = crac.getFlowCnecs().stream().filter(Cnec::isOptimized).collect(Collectors.toSet());
+        for (FlowCnec flowCnec : optimizedFlowCnecs) {
             double margin = computeFlowMargin(flowCnec);
             if (margin < worstMargin) {
                 worstMargin = margin;
```

Comment:
I think you can extract that in another place so you don't have to create a CracResultsHelper to call this method, just like a static method using a Crac and a RaoResult